test/framework/e2e_framework.go

```diff
@@ -269,6 +270,38 @@ func (f *E2eFramework) DeleteDatacenters(namespace string, timeout, interval tim
 	})
 }
 
+// DeleteStargates deletes all Stargates in namespace in all remote clusters.
+// This function blocks until all pods from all Stargates have terminated.
+func (f *E2eFramework) DeleteStargates(namespace string, timeout, interval time.Duration) error {
+	f.logger.Info("deleting all Stargates", "Namespace", namespace)
+
+	for _, remoteClient := range f.remoteClients {
```

Comment:
We are starting to see a pattern across multiple functions that involve performing some action across cluster. Then poll across all clusters until condition is met. Maybe we can extract some of the control flow parts into separate functions. Doesn't have to be done here in this PR. Just thinking out loud...